test/e2e/packagemanifest_e2e_test.go

```diff
@@ -3,170 +3,196 @@ package e2e
-		}
+	AfterEach(func() {
+		cleaner.NotifyTestComplete(GinkgoT(), true)
+	})
 
-		source, err := crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
-		require.NoError(GinkgoT(), err)
-		defer func() {
-			require.NoError(GinkgoT(), crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Delete(context.TODO(), source.GetName(), metav1.DeleteOptions{}))
-		}()
-
-		// Wait for package-server to be ready
-		err = wait.Poll(pollInterval, 1*time.Minute, func() (bool, error) {
-			GinkgoT().Logf("Polling package-server...")
-			_, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
-			if err == nil {
-				return true, nil
-			}
-			return false, nil
+	Context("Given a ConfigMap containing package manifest, CSV and CRD", func() {
+		When("A CatalogSource is created using the ConfigMap as catalog source type", func() {
+			It("Package manifest loading can be verified", func() {
+
+				c := newKubeClient()
+
+				// create a simple catalogsource
+				packageName := genName("nginx")
+				stableChannel := "stable"
+				packageStable := packageName + "-stable"
+				manifests := []registry.PackageManifest{
+					{
+						PackageName: packageName,
+						Channels: []registry.PackageChannel{
+							{Name: stableChannel, CurrentCSVName: packageStable},
+						},
+						DefaultChannelName: stableChannel,
+					},
+				}
+
+				crdPlural := genName("ins")
+				crd := newCRD(crdPlural)
+				catalogSourceName := genName("mock-ocs")
+				namedStrategy := newNginxInstallStrategy(genName("dep-"), nil, nil)
+				csv := newCSV(packageStable, testNamespace, "", semver.MustParse("0.1.0"), []apiextensions.CustomResourceDefinition{crd}, nil, namedStrategy)
+				csv.SetLabels(map[string]string{"projected": "label"})
+				csv.Spec.NativeAPIs = []metav1.GroupVersionKind{{Group: "kubenative.io", Version: "v1", Kind: "Native"}}
+				csvJSON, _ := json.Marshal(csv)
+
+				// Wait for package-server to be ready
+				err := wait.Poll(pollInterval, 1*time.Minute, func() (bool, error) {
+					ctx.Ctx().Logf("Polling package-server...")
+					_, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
+					if err == nil {
+						return true, nil
+					}
+					return false, nil
+				})
+				Expect(err).ToNot(HaveOccurred(), "package-server not available")
+
+				_, cleanupCatalogSource := createInternalCatalogSource(c, crc, catalogSourceName, testNamespace, manifests, []apiextensions.CustomResourceDefinition{crd}, []v1alpha1.ClusterServiceVersion{csv})
+				Expect(err).ToNot(HaveOccurred())
+				defer cleanupCatalogSource()
+
+				_, err = fetchCatalogSource(crc, catalogSourceName, testNamespace, catalogSourceRegistryPodSynced)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("querying the package manifest using the package name", func() {
```

Comment:
IMO it's clearer not to pass the option function to `By`, to reduce nesting and avoid giving the impression of a new container node. The only use case I can imagine for the optional function argument would be calling a named function that was used in multiple specs.